Fixes slimes not having a mischievous face

JohnFulpWillard opened the pull request:
## About The Pull Request

The icon state was improperly named, so it didn't actually exist.
This fixes that.

## Why It's Good For The Game

Fixes a bug

## Changelog

:cl:
fix: Slimes' mischievous emote now works.
/:cl:

JohnFulpWillard committed:
Fixes slimes not having a mischievous face

JohnFulpWillard committed:
Adds a unit test against it

JohnFulpWillard committed:
Revert "Fixes slimes not having a mischievous face"

This reverts commit e4c01166083ca76b2f05858596647155724379e7.

JohnFulpWillard commented on code/modules/mob/living/simple_animal/slime/slime.dm:
```diff
@@ -42,6 +40,17 @@
+	///The current mood of the slime, set randomly or through emotes (if sentient).
+	var/current_mood
+
+	var/AIproc = 0 // determines if the AI loop is activated
+	var/Atkcool = 0 // attack cooldown
+	var/Discipline = 0 // if a slime has been hit with a freeze gun, or wrestled/attacked off a human, they become disciplined and don't attack anymore for a while
+	var/SStun = 0 // stun variable
+
+	var/is_adult = 0
+	var/docile = 0
```
These are all vars I just moved over

san7890 replied:
i hate it here

JohnFulpWillard committed:
Update life.dm

JohnFulpWillard committed:
i hate this

JohnFulpWillard committed:
i didnt run emote properly

JohnFulpWillard committed:
Update slime_mood.dm

JohnFulpWillard committed:
Revert "Revert "Fixes slimes not having a mischievous face""

This reverts commit dc0fe884fd528c0aa4d77d4bd8ccc1f40ccf0338.

JohnFulpWillard committed:
my mistake

JohnFulpWillard committed:
better comment

Zytolg commented:
#73502's CI should work after this if the nuketest decides to stop being a dingus.

Mothblocks commented on code/modules/unit_tests/slime_mood.dm:
```diff
@@ -0,0 +1,11 @@
+		for(var/datum/emote/slime/mood/slime_mood in GLOB.emote_list[key])
+			if(!slime_mood.mood_key)
+				continue
+			TEST_ASSERT(("aslime-[slime_mood.mood_key]" in icon_states(this_guy.icon)), "[slime_mood] is set to give [this_guy] the [slime_mood.mood_key] emote, but the icon state can't be found in [this_guy.icon].")
```
Nit: Would be nice if icon_states was put out into a var since it's surprisingly slow

Mothblocks commented on code/modules/unit_tests/slime_mood.dm:
```diff
@@ -0,0 +1,11 @@
+/datum/unit_test/slime_mood
+
+/datum/unit_test/slime_mood/Run()
+	var/mob/living/simple_animal/slime/this_guy = allocate(/mob/living/simple_animal/slime)
```
This honest to god is not better than `S`

JohnFulpWillard replied:
Get a load of this guy
I've seen a lot of funny names on allocates mobs in unit test and thought why not do the same

san7890 commented on code/modules/mob/living/simple_animal/slime/slime.dm:
```diff
@@ -73,10 +81,30 @@
+		"cerulean",
+		"sepia",
+		"black",
+		"pyrite",
```
alphabetize this list since you've already gone through the trouble of multilining it

JohnFulpWillard replied:
You're dead to me

san7890 replied:
i'm right

JohnFulpWillard committed:
reviews + another minor detail

Slime's 'mutation_table' had defined a var 'slime_mutation', however this is the exact same define as defined on the slime itself, so I renamed it to avoid confusion.

JohnFulpWillard commented on code/modules/mob/living/simple_animal/slime/subtypes.dm:
```diff
@@ -1,79 +1,79 @@
 /mob/living/simple_animal/slime/proc/mutation_table(colour)
-	var/list/slime_mutation[4]
+	var/list/slime_mutation_colors[4]
```
I renamed this because there's literally a define on slimes itself called ``var/list/slime_mutation[4]`` and didn't want people to get confused over it.
Slimes really are fucked up creatures.
